### PR TITLE
VE-2045: add an api call for collecting metrics

### PR DESCRIPTION
**Pull request by Toktar**

### CODE REVIEW

File: libindy/src/api/metrics.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+use indy_api_types::errors::prelude::*;
+use indy_utils::ctypes;
+use indy_api_types::validation::Validatable;
+
+use serde_json;
+use libc::c_char;
+
+/// Generate wallet master key.
+/// Returned key is compatible with "RAW" key derivation method.
+/// It allows to avoid expensive key derivation for use cases when wallet keys can be stored in a secure enclave.
```
Comment (skhoroshavin) [inline libindy/src/api/metrics.rs:16]: I think this needs to be updated

File: libindy/src/api/metrics.rs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+/// err: Error code
+///
+/// #Errors
+/// Common*
+/// Wallet*
+#[no_mangle]
+pub extern fn indy_collect_metrics(command_handle: CommandHandle,
+                                       cb: Option<extern fn(command_handle_: CommandHandle,
+                                                            err: ErrorCode,
+                                                            key: *const c_char)>) -> ErrorCode {
```
Comment (skhoroshavin) [inline libindy/src/api/metrics.rs:28]: This should return list of key-value pairs, not a single string

File: libindy/src/api/metrics.rs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    trace!("indy_collect_metrics: >>> command_handle: {:?}, cb: {:?}",
+           command_handle, cb);
+
+    check_useful_c_callback!(cb, ErrorCode::CommonInvalidParam3);
+
+    trace!("indy_generate_wallet_key: params config: {:?}", secret!(config.as_ref()));
```
Comment (skhoroshavin) [inline libindy/src/api/metrics.rs:34]: I suspect these are also left-overs from code that was taken as a template...

File: libindy/src/commands/metrics.rs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+pub enum MetricsCommand {
+    CollectMetrics(Box<dyn Fn(IndyResult<()>) + Send>)
+}
+
+pub struct MetricsCommandExecutor {
+    close_callbacks: RefCell<HashMap<CommandHandle, Box<dyn Fn(IndyResult<()>)>>>,
+    refresh_callbacks: RefCell<HashMap<CommandHandle, Box<dyn Fn(IndyResult<()>)>>>,
+    // TODO: change  PoolHandle to MetricsHandler
+    open_callbacks: RefCell<HashMap<CommandHandle, Box<dyn Fn(IndyResult<PoolHandle>)>>>,
```
Comment (skhoroshavin) [inline libindy/src/commands/metrics.rs:18]: Do we really use/need these callbacks, or they are also leftovers from wallet command executor?

**Review by skhoroshavin — CHANGES_REQUESTED** [review]

File: libindy/src/commands/metrics.rs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    }
+    fn _collect(&self) -> IndyResult<String> {
+
+        trace!("_collect >>>");
+
+        let metrics_map : HashMap<String, f32> = HashMap::new();
+        let res = serde_json::to_string(&metrics_map).unwrap();
+
+        trace!("_collect <<< res: {:?}", res);
+        debug!("collecting metrics from the current thread");
```
Comment (skhoroshavin) [inline libindy/src/commands/metrics.rs:32]:
```suggestion
        debug!("collecting metrics from command thread");
```

**Review by skhoroshavin — APPROVED** [review]

File: libindy/src/commands/metrics.rs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+            }
+
+        };
+    }
+    fn _collect(&self) -> IndyResult<String> {
+
+        trace!("_collect >>>");
+
+        let metrics_map : HashMap<String, f32> = HashMap::new();
+        let res = serde_json::to_string(&metrics_map).unwrap();
```
Comment (jovfer) [inline libindy/src/commands/metrics.rs:29]:
```suggestion
        let res = serde_json::to_string(&metrics_map)?;
```

**Review by jovfer — CHANGES_REQUESTED** [review]

**Review by jovfer — APPROVED** [review]

**Pull request merged**